### PR TITLE
Update containerd to v1 beta3

### PR DESCRIPTION
**Pull request by cpuguy83**
Closes #35439

### CODE REVIEW

File: hack/dockerfile/binaries-commits [hunk 1]
```diff
@@ -4,7 +4,7 @@ TOMLV_COMMIT=9baf8a8a9f2ed20a8e54160840c492f937eeaf9a
 
 # When updating RUNC_COMMIT, also update runc in vendor.conf accordingly
 RUNC_COMMIT=0351df1c5a66838d0c392b4ac4cf9450de844e2d
```
Comment (thaJeztah) [inline hack/dockerfile/binaries-commits]: Looks like runc was bumped in containerd to `74a17296470088de3805e138d3d87c62e613dfc4` - should we update accordingly?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
left one question/comment

File: hack/dockerfile/binaries-commits (hunk 1 above)
Comment (cpuguy83) [reply hack/dockerfile/binaries-commits]: Bumped to the same version in #35439

**Review by AkihiroSuda — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**